File: ios/RNRtc/RTCVerifyUtil.m
#import "RTCVerifyUtil.h"

#import "RTCBaseModule.h"
#import "RTCConfig.h"

@implementation RTCVerifyUtil

+ (BOOL)isMyself:(int)uid {
    return [RTCBaseBridge mUid] == uid;
}

+ (NSDictionary *)isMyselfStr:(NSString *)uid {
    NSMutableDictionary<NSString *, id> *dic = nil;
    if (uid && uid.length > 0) {
        uid = [uid componentsSeparatedByString:@"_"].firstObject;
        dic = [NSMutableDictionary new];
        dic[KEY_UID] = uid;
        dic[KEY_IS_MYSELF] = @([[RTCBaseBridge mUidStr] isEqualToString:uid]);
    }
    return dic;
}

+ (BOOL)isAvailableUid:(int)uid {
    return UID_DEFAULT != uid;
}

+ (BOOL)isAvailableUidStr:(NSString *)uid {
    return ![[NSString stringWithFormat:@"%d", UID_DEFAULT] isEqualToString:uid];
}

@end

File: android/src/main/java/com/arthas/rtc/RTCVerifyUtil.java
package com.arthas.rtc;

import android.text.TextUtils;

import com.facebook.react.bridge.Arguments;
import com.facebook.react.bridge.WritableMap;

public class RTCVerifyUtil {

    public static boolean isMyself(int uid) {
        return RTCBaseModule.getmUid() == uid;
    }

    public static WritableMap isMyselfStr(String uid) {
        WritableMap map = null;
        if (!TextUtils.isEmpty(uid)) {
            uid = uid.split("_")[0];
            map = Arguments.createMap();
            map.putString(RTCConfig.KEY_UID, uid);
            map.putBoolean(RTCConfig.KEY_IS_MYSELF, RTCBaseModule.getmUidStr().equals(uid));
        }
        return map;
    }

    public static boolean isAvailableUid(int uid) {
        return RTCConfig.UID_DEFAULT != uid;
    }

    public static boolean isAvailableUid(String uid) {
        return !String.valueOf(RTCConfig.UID_DEFAULT).equals(uid);
    }

}
